File: fizzBuzz/fizzBuzz.c
// https://leetcode.com/problems/fizz-buzz/
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
char** fizzBuzz(int n, int* returnSize) {
    *returnSize = n;
    char **arr = (char **)malloc(n * sizeof(char *));

    for (int i = 1; i <= n; i++) {
        arr[i-1] = malloc(9);

        if (i % 15 == 0) {
            strcpy(arr[i-1], "FizzBuzz");
        } else if (i % 5 == 0) {
            strcpy(arr[i-1], "Buzz");
        } else if (i % 3 == 0) {
            strcpy(arr[i-1], "Fizz");
        } else {
            sprintf(arr[i-1], "%d", i);
        }
    }


    return arr;
}

File: fizzBuzz/fizzBuzz.rs
// https://leetcode.com/problems/fizz-buzz/
impl Solution {
    pub fn fizz_buzz(n: i32) -> Vec<String> {
        let mut arr = Vec::with_capacity(n as usize);
        for i in 1..=n {
            if i % 15 == 0 {
                arr.push("FizzBuzz".to_string());
            } else if i % 3 == 0 {
                arr.push("Fizz".to_string());
            } else if i % 5 == 0 {
                arr.push("Buzz".to_string());
            } else {
                arr.push(i.to_string());
            }
        }
        return arr;
    }
}
